accepted/00104-enhanced-ansible-support.md

```diff
@@ -0,0 +1,296 @@
+  module.run:
+    - name: ansible.playbooks
+    - ....
+```
+
+The changed `runplaybook.sls` would then look like:
+```
+run_ansible_playbook:
+  mgrcompat.module_run:
+    - name: ansible.playbooks
+    - playbook: {{ pillar["playbook_path"] }}
+    - rundir: {{ pillar["rundir"] }}
+    - flush_cache: {{ pillar["flush_cache"] }}
+{%- if "inventory_path" in pillar %}
+    - inventory: {{ pillar["inventory_path"] }}
+{% endif %}
+```
+
+#### Displaying targeted systems
+
+Since the list of systems targeted by an ansible playbook can be very long, I propose to display the used inventory as a link in the event history details instead.
+
+In case the selected event details are of the playbook action type we will set an additional request parameter called `inventory` in `SystemHistoryEventAction.java` and render an extra row that would look like:
+```
+Inventory: <link_to_inventory>
+```
+
+Clicking the link would then forward the user to the inventory file in `System` > `Ansible` > `Inventories`.
+
+The inventory then displays the list of systems registered in Uyuni that were targeted by the playbook execution.
+
+Since pre selecting a specific inventory based on the provided URL is not currently supported by the [ansible-path-content.tsx](https://github.com/uyuni-project/uyuni/blob/master/web/html/src/manager/minion/ansible/ansible-path-content.tsx) component, it will have to be adapted to support it.
+The URL to load a specific inventory would look like:
+```
+manager/systems/details/ansible/inventories?sid=<system_id>&iid=<inventory_id>
+```
+with `<system_id>` being the system id of the ansible control node and
+`<inventory_id` being the id of the inventory to load.
+
+## Add UI support to allow editing of variables defined in the playbook
+
+The plan here is to enhance the existing UI in `System > Ansible > Playbooks > select playbook` with generated input fields based on the `vars` defined in the playbook.
+
+For example
+```
+- name: Example playbook with various variable types
+  hosts: all
+  vars:
+    # List variable
+    software_packages:
+      - git
+      - nginx
+      - curl
+
+    # Dictionary variable
+    user_info:
+      username: "devuser"
+      uid: 1001
+      home_dir: "/home/devuser"
+
+    # Integer variable
+    max_open_files: 1024
+
+    # Boolean variable
+    enable_firewall: true
+
+    # String variable
+    welcome_message: "Welcome to the server setup!"
+```
+The UI would generate input fields based on variable type (list, dict, string) that are pre-populated with the default variables set in the playbook (if any).
+Changing the values of the generated input fields will allow users to override the defaults and schedule the playbook using the variables set in said UI.
+
+Additionally we should add a `free-form` text field to allow users to define additonal variables that were not part of the `vars` section and thus don't have an input field generated. This input would have to be in JSON format.
```

Comment:
Instead of free-form, would it be feasible to have some  UI where user can add the variables dynamically , we already know the possible types of the variables . So for user, it will be choosing the right type and then add the values as suitable/needed?

On a general note, in Ansible, YAML is far more commonly used than JSON so my preference would be always to have something that users are used if needed.

Reply:
@parlt91 yaml extra extra_args (the ones from free text) only need to be parsed and merged when passing them to ansbile for execution. You don't want to merge then on the front-end and pass all together to the backend, otherwise, you lose track of what is on the fields, and what is on the free form.